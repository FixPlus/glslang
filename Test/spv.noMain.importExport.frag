#version 460 core

float foo(float, float);

void bar(inout vec3 v, int n);

int bar(vec3 v)
{
    return int(v.x + v.y + v.z);
}

float foo(int n, float s);

float foo(float a, float b)
{
    return a + b;
}
