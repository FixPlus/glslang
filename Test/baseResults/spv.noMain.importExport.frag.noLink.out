spv.noMain.importExport.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:7  Function Definition: bar(vf3; ( global highp int)
0:7    Function Parameters: 
0:7      'v' ( in highp 3-component vector of float)
0:9    Sequence
0:9      Branch: Return with expression
0:9        Convert float to int ( temp int)
0:9          add ( temp highp float)
0:9            add ( temp highp float)
0:9              direct index ( temp highp float)
0:9                'v' ( in highp 3-component vector of float)
0:9                Constant:
0:9                  0 (const int)
0:9              direct index ( temp highp float)
0:9                'v' ( in highp 3-component vector of float)
0:9                Constant:
0:9                  1 (const int)
0:9            direct index ( temp highp float)
0:9              'v' ( in highp 3-component vector of float)
0:9              Constant:
0:9                2 (const int)
0:14  Function Definition: foo(f1;f1; ( global highp float)
0:14    Function Parameters: 
0:14      'a' ( in highp float)
0:14      'b' ( in highp float)
0:16    Sequence
0:16      Branch: Return with expression
0:16        add ( temp highp float)
0:16          'a' ( in highp float)
0:16          'b' ( in highp float)
0:?   Linker Objects
0:?     Linker Function: foo(f1;f1; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( in highp float)
0:?     Linker Function: bar(vf3;i1; ( global void)
0:?       Function Parameters: 
0:?         '' ( inout highp 3-component vector of float)
0:?         '' ( in highp int)
0:?     Linker Function: bar(vf3; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( in highp 3-component vector of float)
0:?     Linker Function: foo(i1;f1; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp int)
0:?         '' ( in highp float)

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 46

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              Source GLSL 460
                              Name 8  "bar(vf3;"
                              Name 7  "v"
                              Name 14  "foo(f1;f1;"
                              Name 12  "a"
                              Name 13  "b"
                              Name 21  "bar(vf3;i1;"
                              Name 25  "foo(i1;f1;"
                              Decorate 14(foo(f1;f1;) Linkage Attributes 678391654 1715155302 15153 0
                              Decorate 21(bar(vf3;i1;) Linkage Attributes 678584674 993224310 3879273 1
                              Decorate 8(bar(vf3;) Linkage Attributes 678584674 993224310 0 0
                              Decorate 25(foo(i1;f1;) Linkage Attributes 678391654 1715155305 15153 1
               2:             TypeFloat 32
               3:             TypeVector 2(float) 3
               4:             TypePointer Function 3(fvec3)
               5:             TypeInt 32 1
               6:             TypeFunction 5(int) 4(ptr)
              10:             TypePointer Function 2(float)
              11:             TypeFunction 2(float) 10(ptr) 10(ptr)
              16:             TypePointer Function 5(int)
              17:             TypeVoid
              18:             TypeFunction 17 4(ptr) 16(ptr)
              22:             TypeFunction 2(float) 16(ptr) 10(ptr)
              26:             TypeInt 32 0
              27:     26(int) Constant 0
              30:     26(int) Constant 1
              34:     26(int) Constant 2
     8(bar(vf3;):      5(int) Function None 6
            7(v):      4(ptr) FunctionParameter
               9:             Label
              28:     10(ptr) AccessChain 7(v) 27
              29:    2(float) Load 28
              31:     10(ptr) AccessChain 7(v) 30
              32:    2(float) Load 31
              33:    2(float) FAdd 29 32
              35:     10(ptr) AccessChain 7(v) 34
              36:    2(float) Load 35
              37:    2(float) FAdd 33 36
              38:      5(int) ConvertFToS 37
                              ReturnValue 38
                              FunctionEnd
  14(foo(f1;f1;):    2(float) Function None 11
           12(a):     10(ptr) FunctionParameter
           13(b):     10(ptr) FunctionParameter
              15:             Label
              41:    2(float) Load 12(a)
              42:    2(float) Load 13(b)
              43:    2(float) FAdd 41 42
                              ReturnValue 43
                              FunctionEnd
 21(bar(vf3;i1;):          17 Function None 18
              19:      4(ptr) FunctionParameter
              20:     16(ptr) FunctionParameter
                              FunctionEnd
  25(foo(i1;f1;):    2(float) Function None 22
              23:     16(ptr) FunctionParameter
              24:     10(ptr) FunctionParameter
                              FunctionEnd
